tests/integration/test_generation.py

```diff
@@ -18,3 +23,39 @@ def test_gene001_simple_callback(dash_duo):
 
     assert dash_duo.wait_for_element("#standard").text == "Standard"
     assert dash_duo.wait_for_element("#nested").text == "Nested"
+
+    dash_duo.percy_snapshot(name="gene001-simple-callback")
+
+
+def test_gene002_arbitrary_resources(dash_duo):
+    app = Dash(__name__)
+
+    app.layout = Div([Button(id="btn"), Div(id="container")])
+
+    @app.callback(Output("container", "children"), [Input("btn", "n_clicks")])
+    def update_container(n_clicks):
+        if n_clicks is None:
+            raise PreventUpdate
+
+        return MyStandardComponent(
+            id="standard", value="Standard", style={"font-family": "godfather"}
+        )
+
+    dash_duo.start_server(app)
+
+    assert (
+        dash_duo.driver.execute_script("return document.fonts.check('1em godfather')")
+        is False
+    )
+
+    dash_duo.wait_for_element("#btn").click()
+    assert dash_duo.wait_for_element("#standard").text == "Standard"
+
+    WebDriverWait(dash_duo.driver, 10).until(
```

Comment:
Is this better than `dash.testing.wait.until`? It's barely any more complicated so fine to keep it, just curious if there's a difference.

Regardless, very nice test!

Reply:
Wasn't aware I could do that. I thought it was either this of `dash_duo._wait_for`, will keep in mind for the future.